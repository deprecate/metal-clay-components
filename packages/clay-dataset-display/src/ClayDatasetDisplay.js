import Component from 'metal-component';
import defineWebComponent from 'metal-web-component';
import Soy from 'metal-soy';
import {Config} from 'metal-state';

import templates from './ClayDatasetDisplay.soy.js';

/**
 * Metal ClayDatasetDisplay component.
 */
class ClayDatasetDisplay extends Component {}

/**
 * State definition.
 * @static
 * @type {!Object}
 */
ClayDatasetDisplay.STATE = {
	/**
	 * CSS classes to be applied to the element.
	 * @instance
	 * @memberof ClayDatasetDisplay
	 * @type {?string|undefined}
	 * @default undefined
	 */
	elementClasses: Config.string(),

	/**
	 * Id to be applied to the element.
	 * @instance
	 * @memberof ClayDatasetDisplay
	 * @type {?string|undefined}
	 * @default undefined
	 */
	id: Config.string(),

	/**
	 * Items of the dataset.
	 * @instance
	 * @memberof ClayDatasetDisplay
	 * @type {?array|undefined}
	 * @default undefined
	 */
	items: Config.array(),

	/**
	 * Flag to indicate if the dataset is selectable.
	 * @instance
	 * @memberof ClayDatasetDisplay
	 * @type {?bool|undefined}
	 * @default false
	 */
	selectable: Config.bool().value(false),

	/**
	 * The path to the SVG spritemap file containing the icons.
	 * @instance
	 * @memberof ClayDatasetDisplay
	 * @type {?string|undefined}
	 * @default undefined
	 */
	spritemap: Config.string(),

	/**
	 * Header of the list.
	 * @instance
	 * @memberof ClayDatasetDisplay
	 * @type {?string|undefined}
	 * @default undefined
	 */
	title: Config.string(),

	/**
	 * Type of the view of the list.
	 * @instance
	 * @memberof ClayDatasetDisplay
	 * @type {?string|undefined}
	 * @default undefined
	 */
	viewType: Config.string(),
};

defineWebComponent('clay-dataset-display', ClayDatasetDisplay);

Soy.register(ClayDatasetDisplay, templates);

export {ClayDatasetDisplay};
export default ClayDatasetDisplay;
