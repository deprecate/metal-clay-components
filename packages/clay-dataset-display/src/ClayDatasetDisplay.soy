{namespace ClayDatasetDisplay}

/**
 * This renders the component's whole content.
 */
{template .render}
	{@param spritemap: string}
	{@param? actionItems: list<[
		disabled: bool,
		href: string,
		icon: string,
		label: string,
		quickAction: bool,
		separator: bool,
		type: string
	]> | ?}
	{@param? elementClasses: string}
	{@param? filterItems: list<[
		checked: bool,
		disabled: bool,
		inputName: string,
		inputValue: string,
		label: string,
		separator: bool,
		type: string
	]>}
	{@param? handleItemToggled_: any}
	{@param? hideFiltersDoneButton: bool}
	{@param? id: string}
	{@param? items: ?}
	{@param? searchActionURL: string}
	{@param? searchFormName: string}
	{@param? searchInputName: string}
	{@param? selectable: bool}
	{@param? sortingOrder: string}
	{@param? title: string}
	{@param? viewType: string}
	{@param? viewTypes: ?}

	{let $attributes kind="attributes"}
		class="clay-dataset-display
			{if $elementClasses}
				{sp}{$elementClasses}
			{/if}
		"

		{if $id}
			id="{$id}"
		{/if}
	{/let}

	<div {$attributes}>
		{if $items}
			{call ClayManagementToolbar.render}
				{param actionItems: $actionItems /}
				{param filterItems: $filterItems /}
				{param hideFiltersDoneButton: $hideFiltersDoneButton /}
				{param searchActionURL: $searchActionURL /}
				{param searchFormName: $searchFormName /}
				{param searchInputName: $searchInputName /}
				{param selectable: $selectable /}
				{param sortingOrder: $sortingOrder /}
				{param spritemap: $spritemap /}
				{param viewTypes: $viewTypes /}
			{/call}

			{delcall ClayDatasetDisplay.List variant="$viewType"}
				{param handleItemToggled_: $handleItemToggled_ /}
				{param items: $items /}
				{param schema: $viewTypes[$viewType].schema /}
				{param selectable: $selectable /}
				{param spritemap: $spritemap /}
				{param title: $title /}
			{/delcall}
		{/if}
	</div>
{/template}

/**
 * Default ClayDatasetDisplay.List implementation. This implementation serves
 * as a mere placeholder for the incr-dom compiler. Tools are expected to
 * provide their own ClayDatasetDisplay.List implementation to activate their
 * UIs and components.
 */
{deltemplate ClayDatasetDisplay.List}
	{@param handleItemToggled_: any}
	{@param items: ?}
	{@param schema: ?}
	{@param spritemap: string}
	{@param? selectable: bool}
	{@param? title: string}
{/deltemplate}

/**
 * This renders a ClayList
 **/
{deltemplate ClayDatasetDisplay.List variant="'list'"}
	{@param handleItemToggled_: any}
	{@param items: ?}
	{@param schema: ?}
	{@param spritemap: string}
	{@param? selectable: bool}
	{@param? title: string}

	<div class="container container-view">
		{call ClayList.render}
			{param events: ['itemToggled': $handleItemToggled_ ] /}
			{param items: $items /}
			{param schema: $schema /}
			{param selectable: $selectable /}
			{param spritemap: $spritemap /}
		{/call}
	</div>
{/deltemplate}