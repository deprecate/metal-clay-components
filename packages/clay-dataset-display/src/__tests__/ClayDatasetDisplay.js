import ClayDatasetDisplay from '../ClayDatasetDisplay';

let component;

describe('ClayDatasetDisplay', function() {
	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('should render the default markup', () => {
		component = new ClayDatasetDisplay();

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayDatasetDisplay with classes', () => {
		component = new ClayDatasetDisplay({
			elementClasses: 'my-custom-class',
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayDatasetDisplay with id', () => {
		component = new ClayDatasetDisplay({
			id: 'myId',
		});

		expect(component).toMatchSnapshot();
	});
});
