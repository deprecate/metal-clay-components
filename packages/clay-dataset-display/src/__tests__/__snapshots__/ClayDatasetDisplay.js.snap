// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayDatasetDisplay should render a ClayDatasetDisplay with classes 1`] = `<div class="clay-dataset-display my-custom-class"></div>`;

exports[`ClayDatasetDisplay should render a ClayDatasetDisplay with id 1`] = `<div class="clay-dataset-display" id="myId"></div>`;

exports[`ClayDatasetDisplay should render the default markup 1`] = `
<div class="clay-dataset-display">
  <nav class="management-bar management-bar-primary navbar navbar-expand-md">
    <div class="container">
      <ul class="navbar-nav">
        <li class="nav-item">
          <div class="custom-control custom-checkbox">
            <label>
              <input checked="" class="custom-control-input" ref="input" type="checkbox">
              <span class="custom-control-indicator"></span>
              <span class="custom-control-description sr-only">Select Page</span>
            </label>
          </div>
        </li>
        <li class="nav-item">
          <span class="navbar-text">1 of 11
            <span class="navbar-breakpoint-down-d-none">items selected</span>
          </span>
        </li>
        <li class="nav-item">
          <button class="btn nav-btn btn-link" type="button">Select all</button>
        </li>
      </ul>
      <ul class="navbar-nav">
        <li class="nav-item navbar-breakpoint-down-d-none">
          <button class="btn quick-action-item btn-unstyled" aria-label="trash" type="button">
            <svg aria-hidden="true" class="lexicon-icon lexicon-icon-trash">
              <title>trash</title>
              <use xlink:href="../../../node_modules/clay/build/images/icons/icons.svg#trash"></use>
            </svg>
          </button>
        </li>
        <li class="nav-item navbar-breakpoint-down-d-none">
          <button class="btn quick-action-item btn-unstyled" aria-label="download" type="button">
            <svg aria-hidden="true" class="lexicon-icon lexicon-icon-download">
              <title>download</title>
              <use xlink:href="../../../node_modules/clay/build/images/icons/icons.svg#download"></use>
            </svg>
          </button>
        </li>
        <li class="nav-item">
          <div class="dropdown dropdown-action">
            <button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn btn-unstyled">
              <svg aria-hidden="true" class="lexicon-icon lexicon-icon-ellipsis-v">
                <title>ellipsis-v</title>
                <use xlink:href="../../../node_modules/clay/build/images/icons/icons.svg#ellipsis-v"></use>
              </svg>
            </button>
            <div class="dropdown-menu">
              <ul class="list-unstyled">
                <li>
                  <a class=" dropdown-item" href="#1">Remove</a>
                </li>
                <li>
                  <a class=" dropdown-item" href="#2">Download</a>
                </li>
                <li aria-hidden="true" class="dropdown-divider" role="presentation"></li>
                <li>
                  <a class=" dropdown-item" href="#3">Edit</a>
                </li>
              </ul>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </nav>
  <div class="container container-no-gutters-sm-down container-view">
    <form>
      <div class="table-responsive">
        <table class="table table-autofit table-hover table-list show-quick-actions-on-hover">
          <thead>
            <tr>
              <th></th>
              <th class="table-cell-expand">
                <button class="btn btn-unstyled" type="button">Name
                  <span class="inline-item inline-item-after">
                    <svg aria-hidden="true" class="lexicon-icon lexicon-icon-order-arrow-up">
                      <title>order-arrow-up</title>
                      <use xlink:href="../../../node_modules/clay/build/images/icons/icons.svg#order-arrow-up"></use>
                    </svg>
                  </span>
                </button>
              </th>
              <th>Origin</th>
              <th>Time</th>
              <th>SpicyLevel</th>
              <th>Download</th>
              <th>Difficulty</th>
            </tr>
          </thead>
          <tbody>
            <tr class="table-divider">
              <td colspan="7">Folders</td>
            </tr>
            <tr>
              <td>
                <div class="custom-control custom-checkbox">
                  <label>
                    <input class="custom-control-input" name="folder" value="1" ref="input" type="checkbox">
                    <span class="custom-control-indicator"></span>
                    <span class="custom-control-description"></span>
                  </label>
                </div>
              </td>
              <td class="table-cell-expand">
                <h4 class="table-list-title">
                  <span class="text-truncate-inline">
                    <span class="text-truncate">Photos</span>
                  </span>
                </h4>
              </td>
              <td>
                <span class="text-truncate"></span>
              </td>
              <td class="text-right"></td>
              <td></td>
              <td>
                <button class="btn btn-sm btn-secondary" type="button">Download</button>
              </td>
              <td>
                <div class="progress-group progress-info">
                  <div class="progress">
                    <div aria-valuemax="100" aria-valuemin="0" aria-valuenow="0" class="progress-bar" role="progressbar" style="width: 0%;"></div>
                  </div>
                  <div class="progress-group-addon">0%</div>
                </div>
              </td>
            </tr>
            <tr>
              <td>
                <div class="custom-control custom-checkbox">
                  <label>
                    <input class="custom-control-input" name="folder" value="2" ref="input" type="checkbox">
                    <span class="custom-control-indicator"></span>
                    <span class="custom-control-description"></span>
                  </label>
                </div>
              </td>
              <td class="table-cell-expand">
                <h4 class="table-list-title">
                  <span class="text-truncate-inline">
                    <span class="text-truncate">Videos</span>
                  </span>
                </h4>
              </td>
              <td>
                <span class="text-truncate"></span>
              </td>
              <td class="text-right"></td>
              <td></td>
              <td>
                <button class="btn btn-sm btn-secondary" type="button">Download</button>
              </td>
              <td>
                <div class="progress-group progress-info">
                  <div class="progress">
                    <div aria-valuemax="100" aria-valuemin="0" aria-valuenow="0" class="progress-bar" role="progressbar" style="width: 0%;"></div>
                  </div>
                  <div class="progress-group-addon">0%</div>
                </div>
              </td>
            </tr>
            <tr class="table-divider">
              <td colspan="7">Recipes</td>
            </tr>
            <tr>
              <td>
                <div class="custom-control custom-checkbox">
                  <label>
                    <input checked="" class="custom-control-input" name="recipe" value="3" ref="input" type="checkbox">
                    <span class="custom-control-indicator"></span>
                    <span class="custom-control-description"></span>
                  </label>
                </div>
              </td>
              <td class="table-cell-expand">
                <h4 class="table-list-title">
                  <span class="text-truncate-inline">
                    <span class="text-truncate">Hamburger</span>
                  </span>
                </h4>
              </td>
              <td>
                <span class="text-truncate">U.S.A</span>
              </td>
              <td class="text-right">10 min.</td>
              <td>
                <span class="label label-success">No Spicy</span>
              </td>
              <td>
                <button class="btn btn-sm btn-secondary" type="button">Download</button>
              </td>
              <td>
                <div class="progress-group progress-info">
                  <div class="progress">
                    <div aria-valuemax="100" aria-valuemin="0" aria-valuenow="2" class="progress-bar" role="progressbar" style="width: 2%;"></div>
                  </div>
                  <div class="progress-group-addon">2%</div>
                </div>
              </td>
            </tr>
            <tr>
              <td>
                <div class="custom-control custom-checkbox">
                  <label>
                    <input class="custom-control-input" name="recipe" value="4" ref="input" type="checkbox">
                    <span class="custom-control-indicator"></span>
                    <span class="custom-control-description"></span>
                  </label>
                </div>
              </td>
              <td class="table-cell-expand">
                <h4 class="table-list-title">
                  <span class="text-truncate-inline">
                    <span class="text-truncate">Ramen</span>
                  </span>
                </h4>
              </td>
              <td>
                <span class="text-truncate">China</span>
              </td>
              <td class="text-right">90 min.</td>
              <td>
                <span class="label label-warning">Spicy</span>
              </td>
              <td>
                <button class="btn btn-sm btn-secondary" type="button">Download</button>
              </td>
              <td>
                <div class="progress-group progress-info">
                  <div class="progress">
                    <div aria-valuemax="100" aria-valuemin="0" aria-valuenow="8" class="progress-bar" role="progressbar" style="width: 8%;"></div>
                  </div>
                  <div class="progress-group-addon">8%</div>
                </div>
              </td>
            </tr>
            <tr>
              <td>
                <div class="custom-control custom-checkbox">
                  <label>
                    <input class="custom-control-input" name="recipe" value="5" ref="input" type="checkbox">
                    <span class="custom-control-indicator"></span>
                    <span class="custom-control-description"></span>
                  </label>
                </div>
              </td>
              <td class="table-cell-expand">
                <h4 class="table-list-title">
                  <span class="text-truncate-inline">
                    <span class="text-truncate">Paella</span>
                  </span>
                </h4>
              </td>
              <td>
                <span class="text-truncate">Spain</span>
              </td>
              <td class="text-right">90 min.</td>
              <td>
                <span class="label label-success">No Spicy</span>
              </td>
              <td>
                <button class="btn btn-sm btn-secondary" type="button">Download</button>
              </td>
              <td>
                <div class="progress-group progress-info">
                  <div class="progress">
                    <div aria-valuemax="100" aria-valuemin="0" aria-valuenow="9" class="progress-bar" role="progressbar" style="width: 9%;"></div>
                  </div>
                  <div class="progress-group-addon">9%</div>
                </div>
              </td>
            </tr>
            <tr>
              <td>
                <div class="custom-control custom-checkbox">
                  <label>
                    <input class="custom-control-input" name="recipe" value="6" ref="input" type="checkbox">
                    <span class="custom-control-indicator"></span>
                    <span class="custom-control-description"></span>
                  </label>
                </div>
              </td>
              <td class="table-cell-expand">
                <h4 class="table-list-title">
                  <span class="text-truncate-inline">
                    <span class="text-truncate">Pizza</span>
                  </span>
                </h4>
              </td>
              <td>
                <span class="text-truncate">Italy</span>
              </td>
              <td class="text-right">30 min.</td>
              <td>
                <span class="label label-success">No Spicy</span>
              </td>
              <td>
                <button class="btn btn-sm btn-secondary" type="button">Download</button>
              </td>
              <td>
                <div class="progress-group progress-info">
                  <div class="progress">
                    <div aria-valuemax="100" aria-valuemin="0" aria-valuenow="6" class="progress-bar" role="progressbar" style="width: 6%;"></div>
                  </div>
                  <div class="progress-group-addon">6%</div>
                </div>
              </td>
            </tr>
            <tr>
              <td>
                <div class="custom-control custom-checkbox">
                  <label>
                    <input class="custom-control-input" name="recipe" value="7" ref="input" type="checkbox">
                    <span class="custom-control-indicator"></span>
                    <span class="custom-control-description"></span>
                  </label>
                </div>
              </td>
              <td class="table-cell-expand">
                <h4 class="table-list-title">
                  <span class="text-truncate-inline">
                    <span class="text-truncate">Carcamusas</span>
                  </span>
                </h4>
              </td>
              <td>
                <span class="text-truncate">Toledo</span>
              </td>
              <td class="text-right">50 min.</td>
              <td>
                <span class="label label-warning">Spicy</span>
              </td>
              <td>
                <button class="btn btn-sm btn-secondary" type="button">Download</button>
              </td>
              <td>
                <div class="progress-group progress-info">
                  <div class="progress">
                    <div aria-valuemax="100" aria-valuemin="0" aria-valuenow="5" class="progress-bar" role="progressbar" style="width: 5%;"></div>
                  </div>
                  <div class="progress-group-addon">5%</div>
                </div>
              </td>
            </tr>
            <tr>
              <td>
                <div class="custom-control custom-checkbox">
                  <label>
                    <input class="custom-control-input" name="recipe" value="8" ref="input" type="checkbox">
                    <span class="custom-control-indicator"></span>
                    <span class="custom-control-description"></span>
                  </label>
                </div>
              </td>
              <td class="table-cell-expand">
                <h4 class="table-list-title">
                  <span class="text-truncate-inline">
                    <span class="text-truncate">Tacos</span>
                  </span>
                </h4>
              </td>
              <td>
                <span class="text-truncate">Mexico</span>
              </td>
              <td class="text-right">30 min.</td>
              <td>
                <span class="label label-danger">Very Spicy</span>
              </td>
              <td>
                <button class="btn btn-sm btn-secondary" type="button">Download</button>
              </td>
              <td>
                <div class="progress-group progress-info">
                  <div class="progress">
                    <div aria-valuemax="100" aria-valuemin="0" aria-valuenow="5" class="progress-bar" role="progressbar" style="width: 5%;"></div>
                  </div>
                  <div class="progress-group-addon">5%</div>
                </div>
              </td>
            </tr>
            <tr class="table-divider">
              <td colspan="7">Chefs</td>
            </tr>
            <tr>
              <td>
                <div class="custom-control custom-checkbox">
                  <label>
                    <input class="custom-control-input" name="chef" value="9" ref="input" type="checkbox">
                    <span class="custom-control-indicator"></span>
                    <span class="custom-control-description"></span>
                  </label>
                </div>
              </td>
              <td class="table-cell-expand">
                <h4 class="table-list-title">
                  <span class="text-truncate-inline">
                    <span class="text-truncate">Ferrán Adriá</span>
                  </span>
                </h4>
              </td>
              <td>
                <span class="text-truncate"></span>
              </td>
              <td class="text-right"></td>
              <td></td>
              <td>
                <button class="btn btn-sm btn-secondary" type="button">Download</button>
              </td>
              <td>
                <div class="progress-group progress-info">
                  <div class="progress">
                    <div aria-valuemax="100" aria-valuemin="0" aria-valuenow="0" class="progress-bar" role="progressbar" style="width: 0%;"></div>
                  </div>
                  <div class="progress-group-addon">0%</div>
                </div>
              </td>
            </tr>
            <tr>
              <td>
                <div class="custom-control custom-checkbox">
                  <label>
                    <input class="custom-control-input" name="chef" value="10" ref="input" type="checkbox">
                    <span class="custom-control-indicator"></span>
                    <span class="custom-control-description"></span>
                  </label>
                </div>
              </td>
              <td class="table-cell-expand">
                <h4 class="table-list-title">
                  <span class="text-truncate-inline">
                    <span class="text-truncate">Gordon Ramsay</span>
                  </span>
                </h4>
              </td>
              <td>
                <span class="text-truncate"></span>
              </td>
              <td class="text-right"></td>
              <td></td>
              <td>
                <button class="btn btn-sm btn-secondary" type="button">Download</button>
              </td>
              <td>
                <div class="progress-group progress-info">
                  <div class="progress">
                    <div aria-valuemax="100" aria-valuemin="0" aria-valuenow="0" class="progress-bar" role="progressbar" style="width: 0%;"></div>
                  </div>
                  <div class="progress-group-addon">0%</div>
                </div>
              </td>
            </tr>
            <tr>
              <td>
                <div class="custom-control custom-checkbox">
                  <label>
                    <input class="custom-control-input" name="chef" value="11" ref="input" type="checkbox">
                    <span class="custom-control-indicator"></span>
                    <span class="custom-control-description"></span>
                  </label>
                </div>
              </td>
              <td class="table-cell-expand">
                <h4 class="table-list-title">
                  <span class="text-truncate-inline">
                    <span class="text-truncate">Alberto Chicote</span>
                  </span>
                </h4>
              </td>
              <td>
                <span class="text-truncate"></span>
              </td>
              <td class="text-right"></td>
              <td></td>
              <td>
                <button class="btn btn-sm btn-secondary" type="button">Download</button>
              </td>
              <td>
                <div class="progress-group progress-info">
                  <div class="progress">
                    <div aria-valuemax="100" aria-valuemin="0" aria-valuenow="0" class="progress-bar" role="progressbar" style="width: 0%;"></div>
                  </div>
                  <div class="progress-group-addon">0%</div>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </form>
  </div>
</div>
`;
